src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -232,7 +232,21 @@ public synchronized void setDeadlineTask(int indexInFilteredList, DeadlineTask n
     }
 
     @Override
+    public synchronized DeadlineTask markDeadlineFinished(int indexInFilteredList)
+            throws IllegalValueException {
+        final DeadlineTask finishedDeadline = taskBook.markDeadlineFinished(getDeadlineTaskSourceIndex(indexInFilteredList));
+        refilterDeadlineTaskFilteredList();
```

Comment:
So later, when we implement the "find tasks by keywords" command, marking a deadline as completed will reset the filter? Would that be what the user expects?